### PR TITLE
implement FromIterator for tuples

### PR DESCRIPTION
**Pull request by debris**
[Rendered](https://github.com/debris/rfcs/blob/from_iterator_tuple/text/0000-from-iterator-tuple.md)

### CODE REVIEW

File: text/0000-from-iterator-tuple.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+fn finished_processing_foo(f: Vec<ProcessedFoo>);
+
+fn finished_processing_bar(b: Vec<ProcessedBar>);
+
+fn old_usage() {
+	let foobar: Vec<_> = (0..10)
+		.into_iter()
+		.map(fetch)
+		.map(|(f, b)| (foo(f), bar(b)))
```
Comment (oli-obk) [inline text/0000-from-iterator-tuple.md]: You can already write this as

```rust
let (processed_foo, processed_bar) = (0..10)
.into_iter()
.map(fetch)
.map(|(f, b)| (foo(f), bar(b)))
.unzip();
```

Comment (debris) [inline text/0000-from-iterator-tuple.md]: I totally missed it! What do you think about `collect` falling back to `unzip`?

Comment (oli-obk) [inline text/0000-from-iterator-tuple.md]: `FromIterator` is used for collections so far. I'm not sure how I feel about changing it to also work for tuples of collections. I'd rather generalize `unzip` in the future over all tuples if we ever get variadic generics.

Comment (debris) [inline text/0000-from-iterator-tuple.md]: ok, thanks for the review! :)

**Pull request closed without merging**